File: backend/src/main/java/com/example/backend/dto/PersonOutputDTO.java
package com.example.backend.dto;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

/**
 * Data Transfer Object for person output.
 */
@Data
@NoArgsConstructor
public class PersonOutputDTO {
    private Long id;
    private String name;
    private Date creationDate;
    private Date lastModifiedDate;
    private Long version;
    private Long personId;
    private boolean hasWheelchair;
    private LocationDTO startLocation;
    private LocationDTO endLocation;
}

File: frontend/src/app/dtos/PersonOutput.dto.ts
import {LocationDto} from "./Location.dto";

export interface PersonOutputDto {
    id: number;
    name: string;
    creationDate: Date;
    lastModifiedDate: Date;
    version: number;
    personId: number;
    startLocation: LocationDto;
    endLocation: LocationDto;
    hasWheelchair: boolean;
    selected?: boolean;
}